apps/src/code-studio/header.js

```diff
@@ -173,21 +190,72 @@ function shareProject() {
 
     ReactDOM.render(
       <Provider store={getStore()}>
-        <ShareDialog
+        <ConnectedShareDialog
           i18n={i18n}
           icon={appOptions.skin.staticAvatar}
           shareUrl={shareUrl}
           isAbusive={dashboard.project.exceedsAbuseThreshold()}
+          isOwner={dashboard.project.isOwner()}
+          isPublished={dashboard.project.isPublished()}
           channelId={dashboard.project.getCurrentId()}
           appType={appType}
           onClickPopup={popupWindow}
           // TODO: Can I not proliferate the use of global references to Applab somehow?
           onClickExport={window.Applab ? window.Applab.exportApp : null}
+          onShowPublishDialog={showPublishDialog}
+          onUnpublish={unpublishProject}
           canShareSocial={canShareSocial}
         />
       </Provider>,
       dialogDom
     );
+
+    getStore().dispatch(showShareDialog());
+  });
+}
+
+const ConnectedPublishDialog = connect(state => ({
+  isOpen: state.header.publishDialog.isOpen,
+  isPublishPending: state.header.publishDialog.isPublishPending,
+  projectId: state.header.publishDialog.projectId,
+  projectType: state.header.publishDialog.projectType,
+}), dispatch => ({
+  onClose: () => dispatch(hidePublishDialog()),
+}))(PublishDialog);
+
+function showPublishDialog() {
+  var publishDialog = document.getElementById('publish-dialog');
+  if (!publishDialog) {
+    publishDialog = document.createElement('div');
+    publishDialog.setAttribute('id', 'publish-dialog');
+    document.body.appendChild(publishDialog);
+  }
+
+  ReactDOM.render(
+    <Provider store={getStore()}>
+      <ConnectedPublishDialog
+        onConfirmPublish={publishProject}
+      />
+    </Provider>,
+    publishDialog
+  );
+
+  const projectId = window.dashboard.project.getCurrentId();
+  const projectType = dashboard.project.getStandaloneApp();
```

Comment:
sadly, header can't import project because project imports header. attempting to do this anyway results in a JS error. I imagine this can probably be cleaned up by having loadApp import header and then pass more callbacks to projects, but I'd rather do that in a separate PR since it seems risky.